lib/commands/skill/add-locales/helper.js

```diff
@@ -0,0 +1,130 @@
+const fs = require('fs-extra');
+const path = require('path');
+const R = require('ramda');
+const async = require('async');
+
+const httpClient = require('@src/clients/http-client');
+const SkillMetadataController = require('@src/controllers/skill-metadata-controller');
+const CliError = require('@src/exceptions/cli-error');
+const ResourcesConfig = require('@src/model/resources-config');
+const Manifest = require('@src/model/manifest');
+const stringUtils = require('@src/utils/string-utils');
+const CONSTANTS = require('@src/utils/constants');
+
+module.exports = {
+    initiateModels,
+    addLocales
+};
+
+/**
+ * Start with loading the project Models global instances.
+ * @param {String} profile
+ */
+function initiateModels(profile) {
+    new ResourcesConfig(path.join(process.cwd(), CONSTANTS.FILE_PATH.ASK_RESOURCES_JSON_CONFIG));
+    const skillPackageSrc = ResourcesConfig.getInstance().getSkillMetaSrc(profile);
+    if (!stringUtils.isNonBlankString(skillPackageSrc)) {
+        throw new CliError('Skill package src is not found in ask-resources.json.');
+    }
+    if (!fs.existsSync(skillPackageSrc)) {
+        throw new CliError(`The skillMetadata src file ${skillPackageSrc} does not exist.`);
+    }
+    const manifestPath = path.join(skillPackageSrc, CONSTANTS.FILE_PATH.SKILL_PACKAGE.MANIFEST);
+    new Manifest(manifestPath);
+}
+
+/**
+ * Add locales by deciding where to get the source of interaction models,
+ * and commit the copy paste from the source to the target location.
+ *
+ * @param {Array} selectedLocales
+ * @param {String} profile
+ * @param {Boolean} doDebug
+ * @param {Function} callback
+ */
+function addLocales(selectedLocales, profile, doDebug, callback) {
+    const iModelSourceByLocales = _resolveSourcesForIModel(selectedLocales, profile, doDebug);
+    const uniqueTemplateSet = new Set(iModelSourceByLocales.values());
+    const iModelFolderPath = path.join(process.cwd(), ResourcesConfig.getInstance().getSkillMetaSrc(profile),
+        CONSTANTS.FILE_PATH.SKILL_PACKAGE.INTERACTION_MODEL, 'custom');
+    // copy iModel JSONs that contains same language
+    iModelSourceByLocales.forEach((v, k) => {
```

Comment:
iModelSourceByLocales is Map but you never use feature of the Map like .get .has for O(1) look up. It is probably cleaner just be array of objects.

Reply:
now it's used in ui.displayAddLocalesResult function we can keep it